### ISSUE TITLE
Operators in Composed GLIDE

### ISSUE THREAD

**Issue by realliyifei** [energy-based-model/Compositional-Visual-Generation-with-Composable-Diffusion-Models-PyTorch#20]
From the paper

> We develop Composed GLIDE (Ours), a version of GLIDE that utilizes our compositional operators to combine textual descriptions, without further training.

Questions:

1. For Composed GLIDE, the operators are trained _only_ on COCO dataset? Or COCO plus CLEVR-obj plus CLEVR-rel as well? (btw the paper didn't mention COCO, which is inconsistent and confused)
2. For Composed GLIDE, we just need to use the trained operators on the off-the-shelf GLIDE text encoder, so only operators should be trained, is this correct?
3. Do you think training the operators on visual genome can improve the result of Composed GLIDE? Have you tried it?

Thanks!

**Comment by nanlliu** [comment]
There might have been some misunderstanding of the paper. Compositional operators are algorithms for test time, so operators can be used on any pretrained conditional diffusion models. The operators themselves have nothing to do with the training.

**Comment by realliyifei** [comment]
Ah yes that's right! Sorry let me rephrase a bit: 

1.  If I want to use Composed GLIDE, I just need to train on the **MS-COCO** by its original images and captions (i.e. pure strings), without the need of training on CLEVR-obj and CLEVR-rel. Is my understanding correct?
2.  Do you think alternatively training on Visual Genome would help Composed GLIDE a bit?
(Note that the paper didn't mention COCO so I am a bit confused)

**Comment by nanlliu** [comment]
1. yeah. clevr-x means a model trained on dataset x.
2. not sure if it would improve, since visual genome annotations are noisy. (not sure why it needs to mention COCO since glide is just a pretrained model provided by openai. details about glide should be directed to the glide paper.)
